cmd/traffic/cmd/manager/internal/state/state.go

```diff
@@ -399,7 +402,7 @@ func (s *State) addClient(sessionID string, client *rpc.ClientInfo, now time.Tim
 	ctx, cancel := context.WithCancel(s.ctx)
 	s.sessions[sessionID] = &clientSessionState{
 		sessionState: sessionState{
-			done:       ctx.Done(),
+			ctx:        ctx,
 			cancel:     cancel,
 			lastMarked: now,
 			dials:      make(chan *rpc.DialRequest),
```

Comment:
Perhaps create a `newSessionState(ctx, lastMarked, dials)` that performs the `ctx, cancel := context.WithCancel(ctx)` internally? Can be reused on line 454.